src/main/java/scala_maven/ScalaMojoSupport.java

```diff
@@ -783,47 +789,89 @@ private String getToolClasspath() throws Exception {
     return options;
   }
 
+  private String getScala3ArtifactId(String a) {
+    String _scalaVersionNStr = _scalaVersionN.toString();
+    // int indexOf_ = _scalaVersionNStr.indexOf("-");
+    return _scalaVersionN.major == 3 ? a + "_" + _scalaVersionNStr : a;
```

Comment:
Same here _scalaVersionNStr should be replaced by the binary version which will be "3" starting from Scala 3.0.0